main.py

```diff
@@ -272,7 +298,10 @@ def infer_worker(gpu, ngpu_per_node, args, res_queue=None):
         dist.init_process_group(backend=args.dist_backend, init_method=args.dist_url,
                                 world_size=args.world_size, rank=args.rank)
 
-    model = build_model(args)
+    model = build_model(args.model, args.rank, args.interval_size, args.afunc,
```

Comment:
for these long argument functions, let's use named arguments to make the usage clearer (and avoid potential bugs).

e.g. `build_model(model=args.modek, rank=args.rank, interval_size=args.interval_size,...)`. Same for the `writer` function below as well

Reply:
we call the writer function using mp.Process. Not sure if I can send in args in the above format ... changed build_model.

Reply:
https://stackoverflow.com/questions/11546858/python-multiprocessing-keyword-arguments I believe describes how to pass keyword args to multi process calls. Can you check if this works in your scenario?

Reply:
changed it, can you check now ?

Reply:
would also appreciate if you can test it quickly to make sure nothing unexpected comes along the way